### PR TITLE
Feature/get app base url configurable cuz hard coded urls are really annoying me today

### LINKED ISSUE #128: Get app URL into settings
I noticed we use a hard-coded "http://" in a lot of places combined with the request host.  I think we need to start tell the app its URL rather than hard-coding scheme and pulling host from the request.  We could just configure the scheme (http or https), or even infer it from the request, but it's going to give us more flexibility if we just make the whole base URL configurable.  If, for instance, somebody wants to proxy the app to be under a sub-path of a base domain (e.g., http://example.com/oni), the simpler setup wouldn't work.  If we ever need offline tasks to know the app's URL for any reason, it has to be configured in full.

### PR DESCRIPTION
**Pull request by jechols**
Closes #128 

There may be problems with this PR, but I recommend getting it merged and filing new tickets to avoid a huge amount of pain with others working on any of the THIRTY-FIVE files I had to change.

### CODE REVIEW

File: core/utils/utils.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 def _rdf_base(request):
     host = request.get_host()
```
Comment (jduss4) [inline core/utils/utils.py]: Is this where the `host` variable that you've been replacing with `settings.BASE_URL` comes from?  If so, is anything else using it, or can it be done away with entirely?

Comment (jduss4) [inline core/utils/utils.py]: Oh wait, it's in _rdf_base, never mind, probably it's not the source of all the other hosts, haha

**Comment by jduss4** [comment]
```
./conf/openoni-canonical.conf:RewriteRule ^/(.*)         http://chroniclingamerica.loc.gov/$1 [redirect,last]
```
^ should that be changed?

**Comment by jechols** [comment]
@jduss4 yes, that's its own problem.  I'm not actually sure what that file is around for.  I'd vote to remove it entirely, but I just don't know what it's for....

**Comment by jduss4** [comment]
Okay, then I shall merge with the greatest of faith in your grepping abilities

**Pull request merged**